File: output/gemma4-e2b/avatar/Java/Python/codeforces_108_B.py
import sys

def solve():
    # Read all input tokens
    data = sys.stdin.read().split()
    
    if not data:
        return

    try:
        n = int(data[0])
    except ValueError:
        return # Handle case where N is not an integer

    if len(data) < n + 1:
        return # Not enough elements provided

    # Read the array A
    a = [int(x) for x in data[1:n+1]]
    
    # Sort the array
    a.sort()
    
    # Check the condition
    for i in range(len(a) - 1):
        # Check if a[i+1] < 2 * a[i] AND a[i] != a[i+1]
        if a[i + 1] < a[i] * 2 and a[i] != a[i + 1]:
            print("YES")
            return
            
    print("NO")

solve()

File: output/gemma4-e2b/avatar/Python/Java/codeforces_108_B.java
import java.util.Arrays;
import java.util.Scanner;

public class codeforces_108_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        if (!scanner.hasNextLine()) {
            return;
        }
        
        String line = scanner.nextLine();
        
        // Split the input string by whitespace to get the numbers
        String[] parts = line.trim().split("\\s+");
        
        if (parts.length < 2) {
            // If there are fewer than 2 elements, the condition cannot be checked, result is NO.
            System.out.println("NO");
            return;
        }

        // Convert string parts to integers
        Integer[] a = new Integer[parts.length];
        try {
            for (int i = 0; i < parts.length; i++) {
                a[i] = Integer.parseInt(parts[i]);
            }
        } catch (NumberFormatException e) {
            // Handle case where input contains non-integer values
            return;
        }
        
        // Sort the array (a.sort())
        Arrays.sort(a);
        
        boolean found = false;
        
        // Iterate through the sorted array (for i in range(1, len(a)))
        for (int i = 1; i < a.length; i++) {
            // Check: a[i] != a[i - 1] and a[i]